### PR TITLE
Add link checking

### PR DESCRIPTION
**Pull request by qwerty287**
Closes https://github.com/woodpecker-ci/woodpecker/issues/3332

### CODE REVIEW

File: .woodpecker/static.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      - pnpx cspell lint --no-progress --gitignore '{**,.*}/{*,.*}'
+
+  - name: prettier
+    image: docker.io/woodpeckerci/plugin-prettier:0.1.0
+    depends_on: []
+    settings:
+      version: 3.2.4
+
+  - name: links
+    image: lycheeverse/lychee
```
Comment (anbraten) [inline .woodpecker/static.yaml:29]:
```suggestion
    image: lycheeverse/lychee:0.14.0
```

File: .woodpecker/static.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+when:
```
Comment (anbraten) [inline .woodpecker/static.yaml:1]: Maybe `static-checks` would be a bit more descriptive as filename

**Comment by pat-s** [comment]
It stopped working when you renamed it from `static.yaml` to `static-checks.yaml`. Maybe the `-` is an issue?
Let's see if `_` works or another naming scheme.

**Comment by anbraten** [comment]
Maybe this one is related:

```
{"level":"error","error":"UNIQUE constraint failed: config.config_hash, config.config_repo_id","time":"2024-02-11T09:45:26Z","message":"failed to find or persist pipeline config for woodpecker-ci/woodpecker"}
{"level":"error","error":"Error #01: failed to find or persist pipeline config for woodpecker-ci/woodpecker\n","ip":"140.82.115.62","latency":1101.056467,"method":"POST","path":"/api/hook","status":500,"time":"2024-02-11T09:45:26Z","user-agent":"GitHub-Hookshot/50103f9","time":"2024-02-11T09:45:26Z"}
```

**Comment by pat-s** [comment]
So not using `_` or `-` seems to work. Seems like there is an issue parsing *new* pipelines which contain either of these?

**Review by 6543 — APPROVED** [review]

**Comment by anbraten** [comment]
We should still fix the error and rename the file to `static-check.yml`

**Pull request merged**